src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -187,14 +190,21 @@ public void InitializeService(ServiceHost serviceHost)
             ConnectionInfo connInfo;
             if (!ConnectionService.TryFindConnection(executeParams.OwnerUri, out connInfo))
             {
-                return requestContext.SendError(SR.QueryServiceQueryInvalidOwnerUri);
+                await requestContext.SendError(SR.QueryServiceQueryInvalidOwnerUri);
+                return;
             }
+            
+            ConnectParams connectParams = new ConnectParams
+            {
+                OwnerUri = executeStringParams.OwnerUri,
+                Connection = connInfo.ConnectionDetails,
+                Type = ConnectionType.Default
+            };
 
-            if (connInfo.ConnectionDetails.MultipleActiveResultSets == null || connInfo.ConnectionDetails.MultipleActiveResultSets == false) {
-                // if multipleActive result sets is not allowed, don't specific a connection and make the ownerURI the true owneruri
-                connInfo = null;
-                executeStringParams.OwnerUri = executeParams.OwnerUri;
-            }
+            await ConnectionService.Connect(connectParams);
```

Comment:
Can this throw? Generally we're wrapping every JSON-RPC message handler with a try/catch where any error results in a requestContext.SendError being sent over the channel. I think we need it here as there's plenty going on that could throw.